staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -233,19 +233,35 @@ type StatefulSetHistoryViewer struct {
 
 // ViewHistory returns a list of the revision history of a statefulset
 // TODO: this should be a describer
-// TODO: needs to implement detailed revision view
 func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision int64) (string, error) {
-	_, history, err := statefulSetHistory(h.c.AppsV1(), namespace, name)
+	sts, history, err := statefulSetHistory(h.c.AppsV1(), namespace, name)
 	if err != nil {
 		return "", err
 	}
 
 	if len(history) <= 0 {
 		return "No rollout history found.", nil
 	}
-	revisions := make([]int64, 0, len(history))
-	for _, revision := range history {
-		revisions = append(revisions, revision.Revision)
+
+	historyInfo := make(map[int64]*appsv1.ControllerRevision)
```

Comment:
The code here is almost the same as `DaemonSetHistoryViewer`. Can we dedup?

Reply:
Hi @janetkuo Thanks for the feedback. I de-duped the print logic in DaemonSetHistoryViewer and StatefulsetHistoryViewer. Please review it.